content/en/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-deprecation-affects-you.md

```diff
@@ -0,0 +1,85 @@
+   infrastructure do not execute docker commands. It might be:
+   - SSH to nodes to troubleshoot;
+   - Node startup scripts;
+   - Monitoring and security agents installed on nodes directly.
+3. Third-party tools that perform above mentioned privileged operations. See
+   [Migrating telemetry and security agents from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents)
+   for more information.
+4. Make sure there is no indirect dependencies on dockershim behavior. Some
+   tooling may be configured to react to Docker-specific behaviors, for example
+   specific logs or metrics. This is an edge case.
+
+## Dependency on Docker explained
+
+A [container runtime](/docs/concepts/containers/#container-runtimes) is software that can
+execute the containers that make up a Kubernetes pod. Kubernetes is responsible for orchestration
+and scheduling of Pods; on each node, the {{< glossary_tooltip text="kubelet" term_id="kubelet" >}}
+uses the container runtime interface as an abstraction so that you can use any compatible
+container runtime.
+
+In its earliest releases, Kubernetes offered compatibility with just one container runtime: Docker.
+Later in the Kubernetes project's history, cluster operators wanted to adopt additional container runtimes.
+The CRI was designed to allow this kind of flexibility - and the kubelet began supporting CRI. However,
+because Docker existed before the CRI specification was invented, the Kubernetes project created an
+adapter component, “dockershim”. The dockershim adapter allows the kubelet to interact with Docker as
+if Docker were a CRI compatible runtime.
+
+You can read about it in [kubernetes Containerd integration goes GA](/blog/2018/05/24/kubernetes-containerd-integration-goes-ga/) blog post.
+
+![Dockershim vs. CRI with Containerd](/images/blog/2018-05-24-kubernetes-containerd-integration-goes-ga/cri-containerd.png)
+
+Switching to Containerd as a container runtime eliminates the middleman. All the
+same containers can be run by container runtimes like Containerd as before. But
+now, since containers were scheduled directly with the container runtime, they
+are not visible to Docker. So any Docker tooling or fancy UI you might have used
+before to check on these containers is no longer available.
+
+You cannot get container information using `docker ps` or `docker inspect`
+commands. As you cannot list containers, you cannot get logs, stop containers,
+or execute something with `docker exec`.
+
+{{< note >}}
+
+Stopping containers via Docker CLI was never a good idea as it is
+better to allow Kubernetes to orchestrate them. So you don’t confuse Kubernetes
+by disappearing containers.
+
+{{< /note >}}
+
+You can still pull images or build them using `docker build` command. But images
+built or pulled by Docker would not be visible to container runtime and
+Kubernetes. They needed to be pushed to some registry to allow them to be used
+by kubernetes.
```

Comment:
I'm not sure who is the audience for this paragraph. Is it people who are running `docker build` directly on
a node?

Reply:
yes. It is not uncommon to create DIY continuous integration tasks or such by running `docker build` on nodes. Linking alternative ways to build containers like https://github.com/GoogleContainerTools/kaniko or [docker buildx suggested in AKS](https://github.com/Azure/AKS/releases/tag/2020-09-21#tag-select-menu-844f6e74-4aee-11eb-80e5-5b2c23eaefc6:~:text=.%20If%20you%20are%20doing%20container%20builds%20in%20cluster%20please%20use%20the%20recommended%20docker%20buildx.) would be very helpful.  And this is one of the next tasks I want to add.